sbi/inference/snpe/snpe_base.py

```diff
@@ -303,11 +273,11 @@ def _train(
         Uses early stopping on a held-out validation set as a terminating condition.
         """
 
-        # get the start index for what training set to use. Either 0 or 1
-        ix = int(self._discard_prior_samples and (round_ > 0))
+        # Starting index for the training set (1 = discard round-0 samples).
+        start = int(self._discard_prior_samples and round_ > 0)
 
         # Get total number of training examples.
-        num_examples = torch.cat(self._theta_bank[ix:]).shape[0]
+        num_examples = sum(map(len, self._theta_bank[start:]))
```

Comment:
add a comment to reduce parsing time for nest?

Reply:
Agreed, I did not change the comment above!
Perhaps the following is more self-documenting, let me know:
``` num_examples = sum(len(round_theta) for round_theta in self._theta_bank[start:])```

Reply:
yes, this is code and comment in one line. 👍